src/main/java/pvws/servlets/PVGetServlet.java

```diff
@@ -0,0 +1,340 @@
+     * 
+     * @param request The HTTP request containing the "pv" parameter.
+     * @param g       The JsonGenerator used to write the JSON response.
+     * @throws IOException If an I/O error occurs while writing the JSON response.
+     */
+    @Override
+    protected void writeJson(final HttpServletRequest request, final JsonGenerator g) throws IOException
+    {
+        // Retrieve the "pv" parameter from the request, which specifies the PV(s) to fetch.
+        String name = request.getParameter("pv");
+
+        // If no PVs are specified, return an error response.
+        if (name == null || name.isEmpty()) {
+            g.writeStartObject();
+            g.writeStringField("name", name);
+            g.writeBooleanField("success", false);
+            g.writeStringField("message", "No PVs have been specified.");
+            g.writeEndObject();
+            return;
+        }
+        
+        // Split the "pv" parameter by commas to handle multiple PVs.
+        String[] pvs = name.split(",");
+        for (int i = 0; i < pvs.length; i++) {
+            pvs[i] = pvs[i].trim(); // Trim whitespace around each PV name
+        }
+        
+        // Handle single PV request.
+        if (pvs.length == 1) {
+            try {
+                // Generate a JSON object for the single PV and write it directly to the response.
+                String jsonObject = generateJsonForPV(pvs[0]);
+                g.writeRawValue(jsonObject); // Write raw JSON string
+            } catch (Exception e) {
+                // If an exception occurs, return an error response for the single PV.
+                g.writeStartObject();
+                g.writeStringField("name", pvs[0]);
+                g.writeBooleanField("success", false);
+                g.writeStringField("message", "An exception occurs when fetching or formatting PV value.");
+                g.writeEndObject();
+            }
+        } else {
+            // Handle multiple PV requests using a thread pool for concurrent processing.
+            int numberOfThreads = 100; // Set a large number of threads for I/O-bound tasks.
+            ExecutorService executorService = Executors.newFixedThreadPool(numberOfThreads);
+
+            // Create a list to store tasks for generating JSON for each PV.
+            List<PVTask> pvTasks = new ArrayList<>();
+
+            // Submit tasks to the thread pool to generate JSON for each PV concurrently.
+            for (int i = 0; i < pvs.length; i++) {
+                int index = i;
+                Future<String> future = executorService.submit(() -> generateJsonForPV(pvs[index]));
+                pvTasks.add(new PVTask(pvs[index], future));
+            }
+
+            g.writeStartArray();
+            for (PVTask pvTask : pvTasks) {
+                try {
+                    // Retrieve the JSON result for the PV from the Future and write it to the response.
+                    String jsonObject = pvTask.future.get();
```

Comment:
You need to add a timeout, can't hang in a servlet forever and keep using resources